### PR TITLE
Fix MultiTrackValidator MVA monitoring

### PR DESCRIPTION
**Pull request by makortel**
This PR fixes a bug introduced in #16559. The logic inferring the track MVA selector names missed a check that the MVA selector is part of the iteration sequence. This omission lead errors for tracking eras without MVA selection (`trackingLowPU`, `trackingPhase1PU70`, `trackingPhase2PU140`) in trackingOnly workflows. The check is added here.

Tested in CMSSW_8_1_X_2016-11-15-2300, no changes expected in non-trackingOnly workflows, or in trackingOnly workflows with run2 or `trackingPhase1` tracking.

@rovere @VinInn

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
type bugfix

**Comment by slava77** [comment]
@cmsbuild please test

the last test failed, it was apparently started before the IB showed up

**Comment by makortel** [comment]
Ping

**Comment by smuzaffar** [comment]
@davidlange6 , this does fix the failing workflows.
```
1325.3_TTbar_13+TTbar_13INPUT+DIGIUP15+RECOUP15_trackingOnlyLowPU+HARVESTUP15_trackingOnly Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Tue Nov 22 11:26:31 2016-date Tue Nov 22 11:21:15 2016; exit: 0 0 0 0
10024.5_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFullINPUT+DigiFull_2017+RecoFull_trackingOnlyPhase1PU70_2017+HARVESTFull_trackingOnly_2017 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Tue Nov 22 11:26:47 2016-date Tue Nov 22 11:21:17 2016; exit: 0 0 0 0
2 2 2 2 tests passed, 0 0 0 0 failed
```

File: Validation/RecoTrack/python/TrackValidation_cff.py [hunk 1]
```diff
@@ -195,15 +195,23 @@ def _getSeedingLayers(seedProducers):
...
     pset = cms.untracked.PSet()
-    for iterName in iterations:
+    for iterName, seqName in _cfg.iterationAlgos(postfix, includeSequenceName=True):
         if hasattr(_iterativeTk_cff, iterName):
             mod = getattr(_iterativeTk_cff, iterName)
+            seq = getattr(_iterativeTk_cff, seqName)
+
+            # Ignore iteration if the MVA selector module is not in the sequence
+            try:
+                print seq.index(mod)
```
Comment (ebrondol) [inline Validation/RecoTrack/python/TrackValidation_cff.py]: @makortel is it supposed to stay? Or was it just a try?

Comment (makortel) [inline Validation/RecoTrack/python/TrackValidation_cff.py]: Thanks Erica, the `print` is **not** supposed to stay. Will fix shortly.

**Comment by makortel** [comment]
@cmsbuild, please test

Removed the printout (thanks @ebrondol!).

**Comment by dmitrijus** [comment]
+1

**Comment by cvuosalo** [comment]
+1

For #16660 18b2b561a9dabd34028cb38d3cda015894a12851

Fix a bug in MVA selection for tracking-only workflows. There should be no change in monitored quantities for other workflows.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_9_0_X_2016-11-21-2300 show no significant differences, as expected. Checking expanded configs with baseline CMSSW_9_0_X_2016-11-21-2300 for workflows 10024.5_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_trackingOnlyPhase1PU70_2017, 23234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D5, and 25202.0_TTbar_13 shows no change at all for the latter two, which aren't tracking-only, while the tracking-only workflow's only change is that MVA labels for the trackValidatorBuildingTrackingOnly are now empty since there is no MVA selection for this workflow.

**Comment by davidlange6** [comment]
+1

**Pull request merged**